From 8a9e9a2c66e90f916c891a80ffe2db0767bd0ae8 Mon Sep 17 00:00:00 2001
From: Matt DeVillier <matt.devillier@gmail.com>
Date: Wed, 10 Dec 2025 11:02:30 -0600
Subject: [PATCH 1/1] mb/lenovo/t480s: Fix headphone jack

Add additional register configuration for the Realtek ALC257 audio
codec on the Lenovo ThinkPad T480s. This includes:

- Hidden register SW reset sequence
- ClassD 2W amplifier configuration
- Jack detection (JD1) setup for headphone port
- Silence data mode threshold setting at -84dB

Copied from the T480, originally sourced from
mb/google/brya/variants/pujjolo/hda_verb.c

Addresses issue #619

Change-Id: I0ddea39b40566d6966e89c77352c0904b3c60da9
Signed-off-by: Matt DeVillier <matt.devillier@gmail.com>
---
 .../sklkbl_thinkpad/variants/t480s/hda_verb.c | 37 ++++++++++++++++++-
 1 file changed, 36 insertions(+), 1 deletion(-)

diff --git a/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480s/hda_verb.c b/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480s/hda_verb.c
index b1d96c5a76..9eb9287f9b 100644
--- a/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480s/hda_verb.c
+++ b/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480s/hda_verb.c
@@ -5,7 +5,7 @@
 const u32 cim_verb_data[] = {
 	0x10ec0257, // Vendor/Device ID: Realtek ALC257
 	0x17aa2258, // Subsystem ID
-	11,
+	18,
 	AZALIA_SUBVENDOR(0, 0x17aa2258),
 
 	AZALIA_PIN_CFG(0, 0x12, AZALIA_PIN_DESC(
@@ -51,6 +51,41 @@ const u32 cim_verb_data[] = {
 		1, 15
 	)),
 
+	//==========Widget node 0x20 - 0 :Hidden register SW reset
+	0x0205001A,
+	0x0204C003,
+	0x0205001A,
+	0x0204C003,
+	0x05850000,
+	0x0584F880,
+	0x05850000,
+	0x0584F880,
+	//==========Widget node 0x20 - 1 : ClassD 2W
+	0x02050038,
+	0x02048981,
+	0x0205001B,
+	0x02040A4B,
+	//==========Widget node 0x20 - 2
+	0x0205003C,
+	0x02043154,
+	0x0205003C,
+	0x02043114,
+	//==========Widget node 0x20 - 3 :
+	0x02050046,
+	0x02040004,
+	0x05750003,
+	0x057409A3,
+	//==========Widget node 0x20 - 4 :JD1 enable 1JD port for HP JD
+	0x02050009,
+	0x02046003,
+	0x0205000A,
+	0x02047770,
+	//==========Widget node 0x20 - 5 : Silence data mode Threshold (-84dB)
+	0x02050037,
+	0x0204FE15,
+	0x02050030,
+	0x02049004,
+
 	0x8086280b, // Vendor/Device ID: Intel Kabylake HDMI
 	0x80860101, // Subsystem ID
 	4,
-- 
2.47.3

